### PR TITLE
K3PO .NET Client

### PR DESCRIPTION
**Pull request by pkhanal**
- Implemented new control protocol (Version 2.0)
- NUnit Extension
- Added an example project

Pending Work:
- Add sample to demonstrate the K3PO test against spec project
- Add script to launch the K3PO server before test and stop the server after test.
- Enable automatic test discover and execution in appveyor.yml to run the K3PO test during build

### CODE REVIEW

**Comment by pkhanal** [comment]
Looking into the build failure.

File: .gitignore [hunk 1]
```diff
@@ -40,4 +40,7 @@ bld/
 [Oo]bj/
 
 # nuget stuffs
-packages/
\ No newline at end of file
+packages/
+
+# Don't ignore this file for now until we find a way to pull nunit.core.interfaces as dependency from nuget
+!nunit.core.interfaces.dll
```
Comment (dpwspoon) [inline .gitignore]: I believe we have standardized on adding newlines at the end of files

**Comment by dpwspoon** [comment]
Can you add instructions on how to setup,  minimal requirements, etc to the README.  When I open the solution I get that its an incompatible with my visual studio version.  Working around that now

**Comment by pkhanal** [comment]
@dpwspoon, updated the README

**Comment by pkhanal** [comment]
@dpwspoon, Thanks!

**Pull request merged**